File: tool/table/client/simplebuff.ts
/*
Author: 
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
*/

module config{

let simplebuff_map = null;
export function simplebuff_map_init(config_obj:Object):void{
	simplebuff_map = config_obj["simplebuff_map"];
}


export class Simplebuff extends Object{
	private m_config:any;
	public static m_static_map:Object = new Object();
	constructor(key){
		super();
		this.m_config = simplebuff_map[key];
		for(let i in this.m_config){
			this[i] = this.m_config[i];
		}
		return;
	}
	public static get_Simplebuff(key){
		if(Simplebuff.m_static_map.hasOwnProperty(key) == false){
			Simplebuff.m_static_map[key] = Simplebuff.create_Simplebuff(key);
		}
		return Simplebuff.m_static_map[key];	
	}
	public static create_Simplebuff(key){
		if(simplebuff_map.hasOwnProperty(key)){
			return new Simplebuff(key);
		}
		return null;	
	}
	public static get_cfg_object():Object
	{
		return simplebuff_map;
	}
}
}

File: app/config/simplebuff.py
# -*- coding: utf-8 -*-

'''
Author: Hannibal
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
'''


simplebuff_map = {};
simplebuff_map[1001] = {"id":1001,"btype":0,"effect":"actor['hpmax']*=1.2","group":1,};
simplebuff_map[1002] = {"id":1002,"btype":0,"effect":"actor['spd']+=20","group":1,};
simplebuff_map[1003] = {"id":1003,"btype":0,"effect":"actor['atk']+=20","group":1,};
simplebuff_map[1004] = {"id":1004,"btype":0,"effect":"actor['def']+=20","group":1,};


class Simplebuff:
	def __init__(self, key):
		config = simplebuff_map.get(key);
		for k, v in config.items():
			setattr(self, k, v);
		return

def create_Simplebuff(key):
		config = simplebuff_map.get(key);
		if not config:
			return
		return Simplebuff(key)

